src/me/StevenLawson/TotalFreedomMod/Commands/Command_nick.java

```diff
@@ -0,0 +1,96 @@
+            final String nickname = args[1];
+            final String nickPlain = ChatColor.stripColor(TFM_Util.colorize(nickname.trim()));
+            String newNickname = nickname.replace("&k", "").replace("&n", "").replace("&m", "").replace("&o", "").replace("&0", "").replace("&", "§");
+            
+            if (!nickPlain.matches(NICKNAME_PATTERN))
+            {
+                playerMsg(ChatColor.RED + "That nickname contains invalid characters.");
+                return true;
+            }
+            if (nickPlain.matches("off"))
+            {
+                TFM_EssentialsBridge.setNickname(player.getName(), null);
+                playerMsg(ChatColor.GOLD + "Removed " + player.getName() + "'s nickname.");
+                player.sendMessage(ChatColor.GOLD + "Your nickname has been removed by " + sender.getName() + ".");
+                return true;
+            }
+            if (player == null)
+            {
+                playerMsg(TFM_Command.PLAYER_NOT_FOUND, ChatColor.RED);
+                return true;
+            }            
+            TFM_EssentialsBridge.setNickname(player.getName(), newNickname);
+            playerMsg(ChatColor.GOLD + "Set " + player.getName() + "'s nickname to " + newNickname + ChatColor.GOLD + ".");
+            player.sendMessage(ChatColor.GOLD + "Your nickname has been changed to " + newNickname + ChatColor.GOLD + " by " + sender.getName() + ".");
+            return true;
+        }
+        if ("off".equals(args[0]))
+        {
+            TFM_EssentialsBridge.setNickname(sender.getName(), null);
+            playerMsg(ChatColor.GOLD + "You no longer have a nickname.");
+            return true;
+        }        
+        final String nickPlain = ChatColor.stripColor(TFM_Util.colorize(args[0].trim()));
+        final String nickInput = args[0];
+        if (!nickPlain.matches(NICKNAME_PATTERN))
+        {
+            playerMsg(ChatColor.RED + "Your nickname contains invalid characters.");
+            return true;
+        }
+        if (nickPlain.length() < 4 || nickPlain.length() > 30)
+        {
+            playerMsg("Your nickname must be between 4 and 30 characters long.");
+            return true;
+        }
+        if (nickInput.contains("&k") || (nickInput.contains("&m") || (nickInput.contains("&n") || (nickInput.contains("&o")) || nickInput.contains("&0"))))
+        {
+            playerMsg(ChatColor.RED + "Your nickname contained forbidden chat formatting codes and as such, the forbidden codes have been removed from your nickname.");          
+        }
+        for (Player player : Bukkit.getOnlinePlayers())
+        {
+            if (player == sender_p)
+            {
+                continue;
+            }
+            if (player.getName().equalsIgnoreCase(nickPlain) || ChatColor.stripColor(player.getDisplayName()).trim().equalsIgnoreCase(nickPlain))
+            {
+                playerMsg("That nickname is already in use.");
```

Comment:
Should we really stop people having identical nicknames? I do understand why you have done it, but would it not be better to check the list of online players and see if they are trying to nick themselves to be another players name, and not another persons nickname?